File: python/codeforces/Tprime.py
import math

def isPrime(n) : 
    if (n <= 1) : 
        return False
    if (n <= 3) : 
        return True

    if (n % 2 == 0 or n % 3 == 0) : 
        return False
  
    i = 5
    while(i * i <= n) : 
        if (n % i == 0 or n % (i + 2) == 0) : 
            return False
        i = i + 6
  
    return True

n = int(input())
x = list(map(int, input().split()))
primes = []

for i in range(n):
    y = math.sqrt(x[i])
    if y == int(y) and (primes.count(y) or isPrime(y)):
        print("YES")
        primes.append(y)
    else:
        print("NO")

File: java/codeforces/Tprime.java
import java.math.BigDecimal;
import java.util.*;

public class Tprime {
    
    public static boolean isPrime(long n) {
        if (n <= 1) { 
            return false;
        }
        if (n <= 3) { 
            return true;
        }
        if (n % 2 == 0 || n % 3 == 0) {
            return false;
        }
    
        long i = 5;
        while(i * i <= n) { 
            if (n % i == 0 || n % (i + 2) == 0) {
                return false;
            }
            i += 6;
        }
    
        return true;
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        BigDecimal decimal;

        int n = sc.nextInt();
        long[] x = new long[n];
        for(int i=0; i<n; i++) {
            x[i] = sc.nextLong();
        }

        for(int i=0; i<n; i++) {
            decimal = new BigDecimal(Math.sqrt(x[i]));
            if(decimal.scale() <= 0 && isPrime(decimal.longValue()))
                System.out.println("YES");
            else
                System.out.println("NO");
        }
        sc.close();
    }
}